File: AD-HOC/PYTHON/TELA_2/1250.py
n = int(input())

for _ in range(n):
    x = int(input())
    cont = 0
    tiro = list(map(int, input().split()))
    pulo = input()

    for j in range(x):
        if pulo[j] == 'S' and tiro[j] < 3:
            cont += 1
        elif pulo[j] == 'J' and tiro[j] > 2:
            cont += 1

    print(cont)

File: AD-HOC/C#/TELA_2/1250.cs
using System;

class Program
{
    static void Main()
    {
        int n, x, i, j, cont;
        char c;

        n = int.Parse(Console.ReadLine());

        for (i = 0; i < n; i++)
        {
            x = int.Parse(Console.ReadLine());
            cont = 0;
            int[] tiro = new int[x];
            char[] pulo = new char[x + 1];

            string input = Console.ReadLine();
            for (j = 0; j < x; j++)
            {
                tiro[j] = int.Parse(input.Split(' ')[j]);
            }

            pulo = Console.ReadLine().ToCharArray();

            for (j = 0; j < x; j++)
            {
                if (pulo[j] == 'S')
                {
                    if (tiro[j] < 3)
                        cont++;
                }
                else
                {
                    if (tiro[j] > 2)
                        cont++;
                }
            }
            
            Console.WriteLine(cont);
        }
    }
}
